src/main/kotlin/io/github/projectmapk/jackson/module/kogera/KotlinModule.kt

```diff
@@ -115,13 +138,35 @@ public class KotlinModule private constructor(
             internal const val DEFAULT_CACHE_SIZE = 512
         }
 
-        public var reflectionCacheSize: Int = DEFAULT_CACHE_SIZE
+        public var initialCacheSize: Int = DEFAULT_CACHE_SIZE
+            private set
+        public var maxCacheSize: Int = DEFAULT_CACHE_SIZE
             private set
 
         private val bitSet: BitSet = KotlinFeature.defaults
 
-        public fun withReflectionCacheSize(reflectionCacheSize: Int): Builder = apply {
-            this.reflectionCacheSize = reflectionCacheSize
+        /**
+         * @throws IllegalArgumentException A value less than [maxCacheSize] was entered for [initialCacheSize].
+         */
+        public fun withInitialCacheSize(initialCacheSize: Int): Builder = apply {
+            checkCacheSize(initialCacheSize, maxCacheSize)
+
+            this.initialCacheSize = initialCacheSize
+        }
+
+        /**
+         * Kogera's internal processing requires a certain cache size.
+         * The lower limit of [maxCacheSize] is set to 16 for extreme use cases,
+         * but it is recommended to set it to 100 or more unless there is a very clear reason.
```

Comment:
Previously, there were no restrictions on the minimum size of the cache, but since there were cases where this did not work properly, restrictions were added.